test/gtest/sigmoid_focal_loss.cpp

```diff
@@ -0,0 +1,301 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "sigmoid_focal_loss.hpp"
+#include "miopen/bfloat16.hpp"
+#include "tensor_holder.hpp"
+#include <miopen/env.hpp>
+
+#define TEST_FWD_REDUCED
+#define TEST_BWD_REDUCED
+#define TEST_FWD_UNREDUCED
+#define TEST_BWD_UNREDUCED
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+
+namespace sigmoidfocalloss {
+struct SigmoidFocalLossForwardTestFloat32 : SigmoidFocalLossFwdTest<float>
+{
+};
+
+struct SigmoidFocalLossForwardTestFloat16 : SigmoidFocalLossFwdTest<half>
+{
+};
+
+struct SigmoidFocalLossForwardTestBFloat16 : SigmoidFocalLossFwdTest<bfloat16>
+{
+};
+
+struct SigmoidFocalLossBackwardTestFloat32 : SigmoidFocalLossBwdTest<float>
+{
+};
+
+struct SigmoidFocalLossBackwardTestFloat16 : SigmoidFocalLossBwdTest<half>
+{
+};
+
+struct SigmoidFocalLossBackwardTestBFloat16 : SigmoidFocalLossBwdTest<bfloat16>
+{
+};
+
+struct SigmoidFocalLossUnreducedForwardTestFloat32 : SigmoidFocalLossUnreducedFwdTest<float>
+{
+};
+
+struct SigmoidFocalLossUnreducedForwardTestFloat16 : SigmoidFocalLossUnreducedFwdTest<half>
+{
+};
+
+struct SigmoidFocalLossUnreducedForwardTestBFloat16 : SigmoidFocalLossUnreducedFwdTest<bfloat16>
+{
+};
+
+struct SigmoidFocalLossUnreducedBackwardTestFloat32 : SigmoidFocalLossUnreducedBwdTest<float>
+{
+};
+
+struct SigmoidFocalLossUnreducedBackwardTestFloat16 : SigmoidFocalLossUnreducedBwdTest<half>
+{
+};
+
+struct SigmoidFocalLossUnreducedBackwardTestBFloat16 : SigmoidFocalLossUnreducedBwdTest<bfloat16>
+{
+};
+}; // namespace sigmoidfocalloss
+
+using namespace sigmoidfocalloss;
+
+#ifdef TEST_FWD_REDUCED
```

Comment:
There is define TEST_FWD_REDUCED right above it.
Please explain why you did this

Reply:
While benchmarking, testing, and debugging we may want to run only a specific set of tests instead of all tests. I added these macros to make it easier to enable/disable all tests of a kernel.